src/viam/sdk/common/exception.cpp

```diff
@@ -0,0 +1,61 @@
+#include <viam/sdk/common/exception.hpp>
+
+namespace viam {
+namespace sdk {
+
+Exception::Exception(const std::string& what)
+    : std::runtime_error("ViamException: " + what), condition_(ErrorCondition::k_general){};
+
+Exception::Exception(ErrorCondition condition, const std::string& what)
+    : std::runtime_error("ViamException: " + what), condition_(condition){};
+
+Exception::~Exception() = default;
+
+std::error_condition Exception::condition() const noexcept {
+    return condition_;
+};
+
+struct ErrorCategory : std::error_category {
```

Comment:
Since you are in `.cpp` scope here, you could just as well pull the implementations inline.

Reply:
Ah good point; threw the struct declaration and its methods in the `make_error_condition` function.